### ISSUE TITLE
@slack/web-api bug, not resolving promise for web.chat.postMessage()

### ISSUE THREAD

**Issue by seanwessmith** [oven-sh/bun#8310]
### What version of Bun is running?

1.0.24+6fa35839c

### What platform is your computer?

Darwin 23.2.0 arm64 arm

### What steps can reproduce the bug?

calling the web.chat.postMessage function will not resolve a promise however the message does successfully get posted to Slack. When running this function in Node the promise resolves as expected.

```
import { WebClient } from "@slack/web-api";

const main = async () => {
  const web = new WebClient(Bun.env.SLACK_TOKEN);
  try {
    console.log("start slack test");
    const response = await web.chat.postMessage({
      channel: Bun.env.SLACK_CHANNEL,
      text: "test",
      blocks: [
        {
          type: "section",
          text: {
            type: "mrkdwn",
            text: "test",
          },
        },
        {
          type: "section",
          text: {
            type: "mrkdwn",
            text: "test",
          },
        },
      ],
    });
    console.log("response from slack test", response);
  } catch (error) {
    console.log("error from slack test", error);
  }
};
main();
```

### What is the expected behavior?

to receive a JSON reply like so

{
  ok: true,
  channel: 'XXXXXX',
  ts: '1705793121.715479',
  message: {
    bot_id: 'XXXXXX',
    type: 'message',
    text: 'test',
    user: 'XXXXXX',
    ts: '1705793121.715479',
    app_id: 'XXXXXX',
    blocks: [ [Object], [Object] ],
    team: 'XXXXXX',
    bot_profile: {
      id: 'XXXXXX',
      app_id: 'XXXXXX',
      name: 'Test Bot',
      icons: [Object],
      deleted: false,
      updated: 1692580490,
      team_id: 'XXXXX'
    }
  },
  response_metadata: { scopes: [ 'chat:write' ], acceptedScopes: [ 'chat:write' ] }
}

### What do you see instead?

no response is received.

**Comment by jozan** [comment]
i encountered another bug related to the same library (issue #8924).

for me `web.chat.postMessage()` works in bun but some other api calls do not.

**Comment by ghost** [comment]
With bun, the `web.chat.postMessage()` works, but the script is killed just after, without any error...
The same script works perfectly with `ts-node`.

I don't understand which part of this library kill bun.

Decided to implement directly raw http call to slack API, and not use their `web-client`.

**Comment by seanwessmith** [comment]
@0xVinCal do you have an example of this raw http call?

**Comment by RJWadley** [comment]
seems like the slack web api is using axios

[this line](https://github.com/slackapi/node-slack-sdk/blob/2649398d61c39349f9c9b6c734d4d7a3a4cfdcc2/packages/web-api/src/WebClient.ts#L221) leads me to think this is caused by issue #4871

**Comment by ghost** [comment]
@seanwessmith 

```typescript

import type { ChatPostMessageResponse } from '@slack/web-api';
import axios from 'axios';
import { config } from '../../config';

/**
 * Send a message to a Slack channel
 */
export async function sendSlackMessage(channel: string, text: string): Promise<ChatPostMessageResponse> {
  const result = await axios.post(
    `https://slack.com/api/chat.postMessage`,
    {
      channel: channel,
      text: text,
    },
    {
      headers: {
        Authorization: `Bearer ${config.slack.apiToken}`,
        Accept: 'application/json',
        'Accept-Encoding': 'identity',
      },
    },
  );

  if (!result.data.ok) {
    throw new Error(`Failed to send Slack message: ${result.data.error}`);
  }

  return result.data;
}


```

**Comment by jozan** [comment]
i took a look at this again. when setting `maxRedirects` to `1` or default by removing the property altogether, i hit another issue: bun doesn't support brotli compression. keep an eye on https://bun.sh/docs/runtime/nodejs-apis#node-zlib.

**Comment by nektro** [comment]
does this still reproduce for you? brotli support was added in [Bun 1.1.8](https://bun.sh/blog/bun-v1.1.8#new-brotli-support-for-node-zlib) and some http improvements made it into [Bun 1.1.9](https://bun.sh/blog/bun-v1.1.9#node-js-compatiblity-improvements).

**Comment by seanwessmith** [comment]
this is now working in [Bun 1.1.9](https://bun.sh/blog/bun-v1.1.9#node-js-compatiblity-improvements)!

```typescript
import type { ChatPostMessageResponse } from "@slack/web-api";
import axios from "axios";

/**
 * Send a message to a Slack channel
 */
export async function sendSlackMessage(
  channel: string,
  text: string
): Promise<ChatPostMessageResponse> {
  const result = await axios.post(
    `https://slack.com/api/chat.postMessage`,
    {
      channel: channel,
      text: text,
    },
    {
      headers: {
        Authorization: `Bearer ${process.env.SLACK_BOT_TOKEN}`,
        Accept: "application/json",
        "Accept-Encoding": "identity",
      },
    }
  );

  if (!result.data.ok) {
    throw new Error(`Failed to send Slack message: ${result.data.error}`);
  }

  return result.data;
}
sendSlackMessage("orders", "Hello, World!").then((response) => {
  console.log(response);
});
```

**Comment by seanwessmith** [comment]
thanks yall

**Comment by jozan** [comment]
tested my repro case and this works in bun 1.1.10
